src/Sentry/GlobalSessionManager.cs

```diff
@@ -53,14 +53,20 @@ private void PersistSession(SessionUpdate update, DateTimeOffset? pauseTimestamp
 
         try
         {
-            Directory.CreateDirectory(_persistenceDirectoryPath);
+            _options.LogDebug("Creating persistence directory for session file at '{0}'.", _persistenceDirectoryPath);
 
-            _options.LogDebug("Created persistence directory for session file '{0}'.", _persistenceDirectoryPath);
+            if (_options.FileSystem.CreateDirectory(_persistenceDirectoryPath))
```

Comment:
~if it returns true it means it failed? seems counter intuitive to me.~

I see below that it does return `false` if it fails, so there's a bug here

```suggestion
            if (!_options.FileSystem.CreateDirectory(_persistenceDirectoryPath))
```